Added Mouse Ops

ps3moira opened the pull request:
Mouse Operator reinforcements are new little buddies to help out with Syndicate goals.

<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
Added Mouse Operatives and a Mouse-Op Reinforcement Radio to the Syndicate Uplink
## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
Added this because Mouse-Ops can aid with their unique size and ability to Syndicate's while also being very affordable.
## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

https://www.youtube.com/watch?v=ybHq5ovoqsk

![Screenshot 2023-11-05 142453](https://github.com/DeltaV-Station/Delta-v/assets/113228053/6e127ac3-ab5c-49af-b2f0-bed3195f8958)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Added mouse operative reinforcements, which can be bought from the Syndicate uplink

ps3moira committed:
Added Mouse Ops

Mouse Operator reinforcements are new little buddies to help out with Syndicate goals.

deltanedas commented:
replace radio with cheese wedge

ps3moira commented:
> replace radio with cheese wedge

I'll make a sprite for a cheese wedge with a tiny radio antenna sticking out or something lol

Adrian16199 commented:
Imagine gettin this and a zombie kit, dear god.

DebugOk commented on Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:
```diff
@@ -17,6 +17,10 @@ deltav-ghost-role-information-salvageantag-rules =  You are a salvage mob.
+# Delta V nukie mouse
+ghost-role-information-nukie-mouse-name = Nuclear Operative Mouse
+ghost-role-information-nukie-mouse-description = A Nuclear Operative reinforcement for the Syndicate.
+
```
Seperate into /en-US/deltav/

ps3moira replied:
Fixed!

DebugOk commented on Resources/Locale/en-US/store/uplink-catalog.ftl:
```diff
@@ -347,3 +347,6 @@ uplink-syndicate-pai-desc = A Syndicate variant of the pAI with access to the Sy
 uplink-bribe-desc = A heartfelt gift that can help you sway someone's opinion. Real or counterfeit? Yes.
+
+uplink-reinforcement-radio-nukie-mouse-name = Mouse-Op Reinforcement Teleporter
+uplink-reinforcement-radio-nukie-mouse-desc = Calls in a specially trained mouse operator to assist you.
```
Seperate into /en-US/deltav/

ps3moira replied:
Done!

ps3moira committed:
Added Moldy Cheese Radio + Moved Locale to DeltaV

VMSolidus commented:
Holy shit this is amazing, I LOVE IT

ps3moira committed:
Removed Inventory Component

deltanedas commented:
besides being given microbombs and eating, what can they *do*

ps3moira commented:
> besides being given microbombs and eating, what can they _do_

I made a little video but basically.

They can have C4 planted on them, insulated so they can chew wires, their suit protects them from space so they can survive spacings, they can talk (not that they can really say much besides squeak) on the Syndicate channel and get orders from the Syndicate channel, and they can do pretty anything a mouse can do while also able to attack/pull things and have a slight armor/damage boost.

deltanedas commented:
real

ps3moira reviewed (commented):
Moved the parts from Locale into the DeltaV folder

ps3moira committed:
You can now pet Mouse-Ops

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

DEATHB4DEFEAT commented:
> **Changelog**
> :cl:
> - add: Added Mouse-Ops
> - add: Added Mouse-Ops Reinforcement Radio
> - add: Added Mouse-Ops Reinforcement Radio to Uplink

Why are there three CLs?
- add: Added mouse operative reinforcements, which can be bought from the Nuclear Operative uplink

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl:
```diff
@@ -0,0 +1,2 @@
+ghost-role-information-nukie-mouse-name = Nuclear Operative Mouse
+ghost-role-information-nukie-mouse-description = A Nuclear Operative reinforcement for the Syndicate.
```
```suggestion
ghost-role-information-nukie-mouse-description = A Nuclear Operative reinforcement for the Syndicate.

```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/interaction/interaction-popup-component.ftl:
```diff
@@ -0,0 +1,7 @@
+## Petting animals
+
+petting-success-nukie-mouse = You pet {THE($target)} on {POSS-ADJ($target)} little blood-red syndicate mouse head.
+
```
```suggestion
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/store/uplink-catalog.ftl:
```diff
@@ -0,0 +1,23 @@
+
+# Misc
+
+# Pointless
```
```suggestion
# Pointless

```

DEATHB4DEFEAT commented on Resources/Locale/en-US/store/uplink-catalog.ftl:
```diff
@@ -346,4 +346,4 @@ uplink-syndicate-pai-name = Syndicate personal ai device
 
 uplink-bribe-name = Lobbying Bundle
-uplink-bribe-desc = A heartfelt gift that can help you sway someone's opinion. Real or counterfeit? Yes.
+uplink-bribe-desc = A heartfelt gift that can help you sway someone's opinion. Real or counterfeit? Yes.
```
```suggestion
uplink-bribe-desc = A heartfelt gift that can help you sway someone's opinion. Real or counterfeit? Yes.

```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/store/uplink-catalog.ftl:
```diff
@@ -0,0 +1,23 @@
+# Ammo
+
+# Utility
+uplink-reinforcement-radio-nukie-mouse-name = Mouse-Op Reinforcement Teleporter
```
```suggestion
uplink-reinforcement-radio-nukie-mouse-name = Mouse Operative Reinforcement Teleporter
```

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/store/uplink-catalog.ftl:
```diff
@@ -0,0 +1,23 @@
+
+# Utility
+uplink-reinforcement-radio-nukie-mouse-name = Mouse-Op Reinforcement Teleporter
+uplink-reinforcement-radio-nukie-mouse-desc = Calls in a specially trained mouse operator to assist you.
```
```suggestion
uplink-reinforcement-radio-nukie-mouse-desc = Calls in a specially trained mouse operative to assist you.
```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Catalog/uplink_catalog.yml:
```diff
@@ -1,3 +1,14 @@
+- type: listing
+  id: Mouse Op Reinforcement
```
```suggestion
  id: Mouse Operative Reinforcement
```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Datasets/Names/mouse_death_commando.yml:
```diff
@@ -0,0 +1,43 @@
+  - Operator La Vache Qui Rit
+  - Operator Mascarpone
+  - Operator Ricotta
+  - Operator Vacherin
```
```suggestion
  - Operator Vacherin

```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/nukiemouse.yml:
```diff
@@ -0,0 +1,164 @@
+    interactFailureString: petting-failure-nukie-mouse
+    interactSuccessSpawn: EffectHearts
+    interactSuccessSound:
+      path: /Audio/Animals/mouse_squeak.ogg
```
```suggestion
      path: /Audio/Animals/mouse_squeak.ogg

```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Entities/Objects/Devices/Syndicate_Gadgets/reinforcement_teleporter.yml:
```diff
@@ -0,0 +1,21 @@
+    sound: /Audio/Animals/mouse_squeak.ogg
+  - type: ItemCooldown
+  - type: UseDelay
+    delay: 300
```
```suggestion
    delay: 300

```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Entities/Objects/Devices/Syndicate_Gadgets/reinforcement_teleporter.yml:
```diff
@@ -0,0 +1,21 @@
+  parent: ReinforcementRadioSyndicate
+  id: ReinforcementRadioSyndicateNukieMouse
+  name: suspicious moldy cheese
+  description: Moldy cheese with a little worm sticking out of it and a...blinking antenna? Calls in a specially trained mouse operator to assist you.
```
```suggestion
  description: Moldy cheese with a little worm sticking out of it and a... blinking antenna? Might attract an odd mouse.
```

DEATHB4DEFEAT commented on Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/nukiemouse.yml:
```diff
@@ -0,0 +1,164 @@
+  name: nuclear operative mouse
+  parent: [ SimpleMobBase, MobCombat ]
+  id: MobNukieMouse
+  description: A nuclear operative...MOUSE?!
```
```suggestion
  description: A Nuclear Operative... MOUSE?!
```

DEATHB4DEFEAT commented on Resources/Textures/DeltaV/Objects/Devices/communication.rsi/meta.json:
```diff
@@ -0,0 +1,25 @@
+		  1.8,
+		  1.8
+        ]
+      ]
```
```suggestion
      "delays": [[ 1.8, 1.8, 1.8, 1.8, 1.8, 1.8, 1.8 ]]
```

ps3moira committed:
Apply suggestions from code review

Co-authored-by: DEATHB4DEFEAT <77995199+DEATHB4DEFEAT@users.noreply.github.com>
Signed-off-by: Ps3Moira <113228053+ps3moira@users.noreply.github.com>

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira commented:
> > **Changelog**
> > 🆑
> > 
> > * add: Added Mouse-Ops
> > * add: Added Mouse-Ops Reinforcement Radio
> > * add: Added Mouse-Ops Reinforcement Radio to Uplink
> 
> Why are there three CLs?
> 
> * add: Added mouse operative reinforcements, which can be bought from the Nuclear Operative uplink

Fixed it

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Update RobustToolbox

ps3moira committed:
Revert "Update RobustToolbox"

This reverts commit 2001a469107f7e4681e1e46f37e5eae6252cd4a4.

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

DEATHB4DEFEAT commented on Resources/Locale/en-US/deltav/store/uplink-catalog.ftl:
```diff
@@ -1,2 +1,25 @@
+
+# Pointless
 uplink-bionic-syrinx-implanter-name = Bionic Syrinx Implanter
 uplink-bionic-syrinx-implanter-desc = An implant that enhances a harpy's natural talent for mimicry to let you adjust your voice to whoever you can think of.
```
These should be moved to another comment category

ps3moira replied:
Where are they supposed to go?

ps3moira replied:
Oh wait, implants. Alright.

ps3moira replied:
Fixed!

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Moved Bionic Syrinx Implant to proper category

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

DebugOk commented on Resources/Locale/en-US/deltav/store/uplink-catalog.ftl:
```diff
@@ -1,2 +1,25 @@
+
+# Ammo
+
+# Utility
```
Why does any of this get added

ps3moira replied:
I removed them.

DebugOk commented on Resources/Locale/en-US/deltav/store/uplink-catalog.ftl:
```diff
@@ -1,2 +1,25 @@
+# Implants
 uplink-bionic-syrinx-implanter-name = Bionic Syrinx Implanter
 uplink-bionic-syrinx-implanter-desc = An implant that enhances a harpy's natural talent for mimicry to let you adjust your voice to whoever you can think of.
+
```
Same deal here

ps3moira replied:
That's a Delta-V unique syndicate item, that's why they got placed there and because it's a implant.

DebugOk replied:
I'm talking about everything below that, for some reason it selected the wrong lines

ps3moira replied:
I can remove it

ps3moira replied:
I can remove it

DebugOk commented on Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/nukiemouse.yml:
```diff
@@ -0,0 +1,164 @@
+    - Syndicate
+  # make the player a traitor once its taken
+  - type: AutoTraitor
+    giveUplink: false
```
Ghost roles shouldn't be traitors, especially not those purchased. Reinforcements dont to this either, so a mouse especially shouldn't.

Just add them to the syndicate faction.
```
- type: NpcFactionMember
  factions:
    - Syndicate
 ```

ps3moira replied:
It does for Syndicate  Reinforcement Monkey's 
![image](https://github.com/DeltaV-Station/Delta-v/assets/113228053/bc5fc0b0-4353-4bb4-8662-9ffcaec75458)

ps3moira replied:
So I'd say it's fine for mouse-ops, who can do a lot less than a monkey can do

ps3moira replied:
> Ghost roles shouldn't be traitors, especially not those purchased. Reinforcements dont to this either, so a mouse especially shouldn't.
> 
> Just add them to the syndicate faction.
> 
> ```
> - type: NpcFactionMember
>   factions:
>     - Syndicate
> ```

I also added this already.

DebugOk commented on Resources/Prototypes/DeltaV/Entities/Objects/Devices/Syndicate_Gadgets/reinforcement_teleporter.yml:
```diff
@@ -0,0 +1,21 @@
+    layers:
+    - state: cheese-radio
+  - type: GhostRole
+    name: Nuclear Mouse Operator Reinforcement
+    description: Someone needs reinforcements. You, a trained mouse, will help them.
+    rules: Normal syndicate antagonist rules apply. Work with whoever called you in, and don't harm them.
```
These should be localized.

ps3moira replied:
How would I do that?

ps3moira replied:
I was able to localize them in the fixed branch https://github.com/DeltaV-Station/Delta-v/pull/498

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Removed Unused Catagories

ps3moira committed:
Revert "Merge remote-tracking branch 'upstream/master' into mouse-ops"

This reverts commit c36396b9d34be09f550a9ef0b25eebc0e1ed21fe, reversing
changes made to a6027bd76d98e02b00f0df4cb08468afb3fa57a3.

ps3moira committed:
Revert "Revert "Merge remote-tracking branch 'upstream/master' into mouse-ops""

This reverts commit 8f4999fb7e61a01e9d729e64d81ae679326c24dc.

ps3moira committed:
Update uplink-catalog.ftl

ps3moira committed:
Merge remote-tracking branch 'upstream/master' into mouse-ops

ps3moira committed:
Merge branch 'master' into buff-mouse

ps3moira committed:
Merge branch 'master' into buff-mouse

ps3moira committed:
Merge branch 'master' into buff-mouse

ps3moira committed:
Merge branch 'master' into buff-mouse

ps3moira commented:
I closed this branch due to me accidentally messing up the code with robust tools. Bah humbug. Anyway, it's fixed, localized and ported to here https://github.com/DeltaV-Station/Delta-v/pull/498
